src/hooks/useTikTokAuth.ts

```diff
@@ -0,0 +1,155 @@
+    const expiration = window.localStorage.getItem(
+      'tikTokAccessTokenExpiration'
+    )
+
+    const isExpired = expiration && moment().isAfter(expiration)
+    if (accessToken && openId && !isExpired) {
+      callback(accessToken, openId)
+    } else {
+      getRequestToken(callback)
+    }
+  }
+
+  const getRequestToken = (callback: WithAuthCallback) => {
+    const popup = openPopup()
+
+    if (popup) {
+      const authenticationUrl = `${IDENTITY_SERVICE}/tikTok`
+
+      popup.location.href = authenticationUrl
+      poll(popup, callback)
+    }
+  }
+
+  const openPopup = () => {
+    return window.open(
+      '',
+      '',
+      'toolbar=no, location=no, directories=no, status=no, menubar=no, scrollbars=no, resizable=no, copyhistory=no, width=600, height=400'
+    )
+  }
+
+  const poll = (popup: Window, callback: WithAuthCallback) => {
+    const interval = setInterval(async () => {
+      if (!popup || popup.closed || popup.closed === undefined) {
+        clearInterval(interval)
+        onError(new Error('Popup has been closed by user'))
+        return
+      }
+
+      const closeDialog = () => {
+        clearInterval(interval)
+        popup.close()
+      }
+
+      try {
+        if (
+          !popup.location.hostname.includes('tiktok.com') &&
+          popup.location.hostname !== ''
+        ) {
+          if (popup.location.search) {
+            const query = new URLSearchParams(popup.location.search)
+
+            const authorizationCode = query.get('code')
+            const csrfState = query.get('state')
+            const error = query.get('error')
+            if (authorizationCode && csrfState) {
+              closeDialog()
+              return await getAccessToken(
+                authorizationCode,
+                csrfState,
+                callback
+              )
+            } else {
+              closeDialog()
+              return onError(
+                new Error(error || 'An error occured during OAuth')
+              )
+            }
+          } else {
+            closeDialog()
+            return onError(
+              new Error(
+                'OAuth redirect has occurred but no query or hash parameters were found. ' +
+                  'They were either not set during the redirect, or were removed—typically by a ' +
+                  'routing library—before Twitter react component could read it.'
+              )
+            )
+          }
+        }
+      } catch (error) {
+        // Ignore DOMException: Blocked a frame with origin from accessing a cross-origin frame.
+        // This will catch until the popup is redirected back to the same origin
+      }
+    }, 500)
+  }
+
+  const getAccessToken = async (
+    authorizationCode: string,
+    csrfState: string,
+    callback: WithAuthCallback
+  ) => {
+    try {
+      const response = await window.fetch(
+        `${IDENTITY_SERVICE}/tikTok/access_token`,
```

Comment:
nit, it looks like we've got mixed camelCase and snake_case endpoints already, but within a single endpoint, let's make consistent

Reply:
Fixed! Turns out `tiktok` was all lower on the backend anyway